src/target/riscv/riscv-013.c

```diff
@@ -3424,114 +3412,189 @@ static void log_mem_access_result(struct target *target, bool success, int metho
 		LOG_TARGET_DEBUG(target, "%s", msg);
 }
 
-static bool mem_should_skip_progbuf(struct target *target, target_addr_t address,
-		uint32_t size, bool is_read, char **skip_reason)
+typedef enum {
+	MEM_ACCESS_STATUS_KIND_OK,
+	MEM_ACCESS_STATUS_KIND_DISABLED,
+	MEM_ACCESS_STATUS_KIND_SKIPPED,
+	MEM_ACCESS_STATUS_KIND_FAILED
+} mem_access_status_kind_t;
+
+#define LIST_OF_MEM_ACCESS_STATUSES                                                                \
+	MEM_ACCESS_STATUS_HANDLER(MEM_ACCESS_STATUS_OK,                                            \
+			MEM_ACCESS_STATUS_KIND_OK, "ok")                                           \
+	MEM_ACCESS_STATUS_HANDLER(MEM_ACCESS_STATUS_DISABLED,                                      \
+			MEM_ACCESS_STATUS_KIND_DISABLED, "disabled")                               \
+	MEM_ACCESS_STATUS_HANDLER(MEM_ACCESS_STATUS_SKIPPED_ABSTRACT_ACCESS_CMDERR,                \
+			MEM_ACCESS_STATUS_KIND_SKIPPED, "skipped (abstract access cmderr)")        \
```

Comment:
If you'd like to shorten the identifiers, you can make the prefix shorter `MEM_ACCESS_STATUS_` --> `MEM_ACCESS_` and they still remain very clear and readable.

```suggestion
	MEM_ACCESS_STATUS_HANDLER(MEM_ACCESS_OK,                                            \
			MEM_ACCESS_STATUS_KIND_OK, "ok")                                           \
	MEM_ACCESS_STATUS_HANDLER(MEM_ACCESS_DISABLED,                                      \
			MEM_ACCESS_STATUS_KIND_DISABLED, "disabled")                               \
	MEM_ACCESS_STATUS_HANDLER(MEM_ACCESS_SKIPPED_ABSTRACT_ACCESS_CMDERR,                \
			MEM_ACCESS_STATUS_KIND_SKIPPED, "skipped (abstract access cmderr)")        \
```

Reply:
Addressed